startupscript/aws/post-startup-hook.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash -x
+
+# post-startup-hook.sh
+#
+# Performs additional one-time configuration for applications running in AWS
+# EC2 instances.
+#
+# Note that this script is intended to be sourced from the "post-startup.sh"
+# script and is dependent on some functions and variables already being set up
+# and some packages already installed:
+#
+# - aws (cli from ghcr.io/devcontainers/features/aws-cli:1)
+# - emit (function)
+# - RUN_AS_LOGIN_USER: run command as app user
+# - USER_BASH_PROFILE: path to user's ~/.bash_profile file
+# - USER_BASHRC: path to user's ~/.bashrc file
+# - USER_NAME: name of app user
+# - USER_PRIMARY_GROUP: name of primary group app user belongs to
+# - USER_PROFILE: path to user's ~/.profile file
+# - USER_WORKBENCH_CONFIG_DIR: user's WB configuration directory
+# - WORKBENCH_GIT_REPOS_DIR: path to the git repo directory (~/repos)
+# - WORKBENCH_INSTALL_PATH: path to CLI executable
+
+readonly AWS_VAULT_INSTALL_PATH="/usr/bin/aws-vault"
+
+##########################################
+# Install aws-vault for credential caching
+##########################################
+emit "installing aws-vault"
+curl --no-progress-meter --location --output "${AWS_VAULT_INSTALL_PATH}" \
+    https://github.com/99designs/aws-vault/releases/download/v7.2.0/aws-vault-linux-amd64
+chmod 755 "${AWS_VAULT_INSTALL_PATH}"
+
+#####################################
+# Set up aws-vault credential caching
+#####################################
+${RUN_AS_LOGIN_USER} "${WORKBENCH_INSTALL_PATH} config set cache-with-aws-vault true"
+${RUN_AS_LOGIN_USER} "${WORKBENCH_INSTALL_PATH} config set wb-path --path ${WORKBENCH_INSTALL_PATH}"
+${RUN_AS_LOGIN_USER} "${WORKBENCH_INSTALL_PATH} config set aws-vault-path --path ${AWS_VAULT_INSTALL_PATH}"
+
+###############################################################################
+# Write common environment vars to .profile so they are present in all contexts
+###############################################################################
+readonly WORKBENCH_WORKSPACE=$(get_metadata_value terra-workspace-id)
+readonly AWS_CONFIG_FILE="${USER_WORKBENCH_CONFIG_DIR}/aws/${WORKBENCH_WORKSPACE}.conf"
+cat >> "${USER_PROFILE}" << EOM
+### BEGIN: Workbench AWS-specific customizations ###
+export WORKBENCH_WORKSPACE="${WORKBENCH_WORKSPACE}"
+export WORKBENCH_GIT_REPOS_DIR="${WORKBENCH_GIT_REPOS_DIR}"
+export WORKBENCH_INSTALL_PATH="${WORKBENCH_INSTALL_PATH}"
+export AWS_CONFIG_FILE="${AWS_CONFIG_FILE}"
+export AWS_VAULT_BACKEND="file"
+export AWS_VAULT_FILE_PASSPHRASE=""
+EOM
+chown "${USER_PRIMARY_GROUP}:${USER_NAME}" "${USER_PROFILE}"
+
+#######################################
+# Write VWB helper functions to .bashrc
+#######################################
+cat >> "${USER_BASHRC}" << 'EOM'
+
+configure_workspace() {
```

Comment:
Have encouraged the pattern for functions:

```
function my_fn() {
...
}
readonly -f my_fn
```

Slightly more verbose, but makes the code a bit more readable and marking functions readonly prevents an unpleasant category of bugs from cropping up.